### PR TITLE
Add sandboxing setting and test

### PR DESCRIPTION
**Pull request by The-Compiler**
This does a bunch of related things:

- Renames `qt.low_end_device_mode` -> `qt.chromium.low_end_device_mode` and `qt.process_model` -> `qt.chromium.process_model` to keep Chromium settings a bit more separated from QtWebEngine ones.
- Adds a new `qt.chromium.sandboxing` setting to easily disable/reduce sandboxing, which is often necessary when trying an older Qt on a newer glibc - while possible with `--qt-flag`, using `-s` is easier and catches typos and such.
- Changes `testutils.seccomp_args` into a `testutils.disable_seccomp_bpf_sandbox` which just returns a boolean, because the sandboxing tests needs it too
- Changes `testutils.disable_seccomp_bpf_sandbox` to get the proper QtWebEngine version via `version.qtwebengine_versions` (rather than its own guessing)
- Changes `testutils.disable_seccomp_bpf_sandbox` to disable the BPF sandbox during testing on some more Qt versions I've had trouble with (probably since a recent libc upgrade?)
- Adds a new `test_sandboxing` end to end test, which checks `chrome://sandbox` and does an exact match on its contents. Might need some fine-tuning and such, but I think it's useful for it to be strict, to also catch issues where sandboxing wouldn't work as expected in a new QtWebEngine or something.

### CODE REVIEW

File: doc/help/settings.asciidoc [hunk 1]
```diff
@@ -3772,6 +3773,73 @@ Type: <<types,List of String>>
...
+Default: +pass:[process-per-site-instance]+
+
+[[qt.chromium.sandboxing]]
+=== qt.chromium.sandboxing
+What sandboxing mechanisms in Chromium to use.
+Chromium has various sandboxing layers, which should be enabled for normal browser usage. Mainly for testing and development, it's possible to disable individual sandboxing layers via this setting.
+Open `chrome://sandbox` to see the current sandbox status.
+Changing this setting is only recommended if you know what you're doing, as it **disables one of Chromium's security layers**. To avoid sandboxing being accidentally disabled persistently, this setting can only be set via `config.py`, not via `:set`.
+See the Chromium documentation for more details:
+- https://chromium.googlesource.com/chromium/src/+/HEAD/docs/linux/sandboxing.md[Linux] - https://chromium.googlesource.com/chromium/src/+/HEAD/docs/design/sandbox.md[Windows] - https://chromium.googlesource.com/chromium/src/+/HEAD/docs/design/sandbox_faq.md[FAQ (Windows-centric)]
```
Comment (The-Compiler) [inline doc/help/settings.asciidoc:3825]: This is kinda broken, but meh, we want to switch away from asciidoc anyways.

File: doc/qutebrowser.1.asciidoc [hunk 2]
```diff
@@ -34,7 +34,7 @@ show it.
 *'URL'*::
     URLs to open on startup (empty as a window separator).
 
-=== optional arguments
+=== options
```
Comment (The-Compiler) [inline doc/qutebrowser.1.asciidoc:37]: No idea why this changed, let's see if the CI is happy with it. Maybe different Python versions.

File: qutebrowser/config/configdata.yml [hunk 3]
```diff
@@ -284,6 +290,39 @@ qt.low_end_device_mode:
...
+    accidentally disabled persistently, this setting can only be set via
+    `config.py`, not via `:set`.
+
+    See the Chromium documentation for more details:
+
+    - https://chromium.googlesource.com/chromium/src/+/HEAD/docs/linux/sandboxing.md[Linux]
+    - https://chromium.googlesource.com/chromium/src/+/HEAD/docs/design/sandbox.md[Windows]
+    - https://chromium.googlesource.com/chromium/src/+/HEAD/docs/design/sandbox_faq.md[FAQ (Windows-centric)]
+  # yamllint enable rule:line-length
+
```
Comment (The-Compiler) [inline qutebrowser/config/configdata.yml:325]: @mschilli87 do your thing please :slightly_smiling_face:

File: tests/helpers/testutils.py [hunk 4]
```diff
@@ -267,35 +262,38 @@ def easyprivacy_txt():
...
+    except ImportError:
+        # no QtWebEngine available
+        return False
+
     affected_versions = set()
     for base, patch_range in [
-            # 5.12.0 to 5.12.7 (inclusive)
-            ('5.12', range(0, 8)),
+            # 5.12.0 to 5.12.10 (inclusive)
+            ('5.12', range(0, 11)),
```
Comment (The-Compiler) [inline tests/helpers/testutils.py:284]: Guess we could also base it on the glibc version or something, but probably not worth the effort, especially if we're gonna drop 5.12 soon.

File: qutebrowser/config/configdata.yml (hunk 3 above)
Comment (mschilli87) [reply qutebrowser/config/configdata.yml:325]: LGTM, but why mixing enable/disable options? AFAICT `qt.chromium.sandboxing.disable`  with options `none`/`seccomp-bpf`/`all` would provide the same choices in a more consistent way. If you decide to go for it, I'd rephrase the description of `none` to `Keep all available sandboxing mechanisms enabled`.

Comment (The-Compiler) [reply qutebrowser/config/configdata.yml:325]: I like using positive names to avoid the mental gymnastics of having a double negative. I don't remember where I picked it up, but the *Use positive, “happy” names* part of [Common API mistakes and how to avoid them - LogRocket Blog](https://blog.logrocket.com/common-api-mistakes-and-how-to-avoid-them-804fbcb9cc4b/) seems to summarize it well.

Even if a fully enabled sandbox is the obvious default, I think it's (slightly) easier to parse `sandboxing = "enable-all"` or `sandboxing = disable-all"` than it is to parse `"sandboxing.disable = "none"` or `sandboxing.disable = "all"`

Comment (mschilli87) [reply qutebrowser/config/configdata.yml:325]: For me the idea was to 'warn' the user that all (non-default) choices for this option *decrease* security. If I were a new user and I'd see an option 'disable safety-features', I'd avoid it. If it was just named 'safety features' and contained both 'enable ...' and 'disable ...' choices, I'd be intrigued. But I understood this is a question of opinion and I can follow your argument as well. That's why I just asked instead of suggesting a change in my review in the first place. So no need to bikeshed and thanks for sharing your reasoning.

**Review by toofar — APPROVED** [review]
LGTM.

I don't understand where the "\n\n\n" thing is coming from. I'm looking at
5.12.8 in a chroot and it looks like a table :shrug:

File: tests/end2end/test_invocations.py [hunk 5]
```diff
@@ -847,6 +855,12 @@ def test_sandboxing(
     quteproc_new.open_url('chrome://sandbox')
     text = quteproc_new.get_content()
     print(text)
+
+    not_found_msg = ("The webpage at chrome://sandbox/ might be temporarily down or "
+                     "it may have moved permanently to a new web address.")
```
[the remark this answers is not in the document]
Comment (The-Compiler) [reply tests/end2end/test_invocations.py:859]: Windows and macOS - it looks like [QtWebEngine only supports](https://codereview.qt-project.org/c/qt/qtwebengine/+/205250) `chrome://sandbox` on Linux, probably because on Windows the page looks different (I'm guessing an entirely different code path):

image

(can't test macOS right now, for whatever reason, my Mac doesn't seem to be reachable via VNC anymore...).

So I suppose if this changes in the future, the test will start failing, and we will notice.

File: doc/help/settings.asciidoc (hunk 1 above)
Comment (The-Compiler) [reply doc/help/settings.asciidoc:3825]: Good catch. Asciidoc is weird. I used `\+` on all of them now, which seems to help. I think having them clickable in the docs is more important than having them clickable in the asciidoc source.

**Review by The-Compiler — COMMENTED** [review]
> I don't understand where the "\n\n\n" thing is coming from. I'm looking at
5.12.8 in a chroot and it looks like a table

It is a table, it's just that `.get_content()` gives us a plaintext version and we parse that. Apparently between Qt versions, that representation changed (or maybe the page source changed). We could parse the HTML in the test instead to reconstruct the table as a dict, but I thought this would be easier (and we'll soon get rid of Qt 5.12 anyways I suppose).

**Comment by The-Compiler** [comment]
Merged after an amend to fix the docs - thanks @toofar and @mschilli87!
